### ISSUE TITLE
Wrong warning "You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`"

### ISSUE THREAD

**Issue by Vadorequest** [zeit/next.js#12476]
# Bug report

## Describe the bug

> Warning: You have opted-out of Automatic Static Optimization due to `getInitialProps` in `pages/_app`. This does not opt-out pages with `getStaticProps`

![image](https://user-images.githubusercontent.com/3807458/80979045-242cb580-8e27-11ea-908f-6965701ab52e.png)

My `_app` contains a **commented/disabled** `getInitialProps`.

```tsx
// static async getInitialProps(props: AppInitialProps): Promise<AppRenderProps> {
```

## To Reproduce

Use a commented `getInitialProps` in _app

## Expected behavior

Warning shouldn't appear when `getInitialProps` isn't actually used

## Screenshots

Video (1mn): 
https://youtu.be/4J6kxF8igIc

**Comment by ijjk** [comment]
Hi, it looks like you are using a HOC that is adding `getInitialProps` to your `_app` so the warning for opting out of the automatic static optimization does appear to be correct

**Comment by Vadorequest** [comment]
That's surprising me. I've just performed a deep find and there are only 6 results:

![image](https://user-images.githubusercontent.com/3807458/81008669-987d4e00-8e53-11ea-8bd0-8e767cee507f.png)

Are _document and _error files affecting this?

You mention an HOC, but only use one in the _app file which doesn't seem to define a `getInitialProps`:

```tsx

import { getDataFromTree } from '@apollo/react-ssr';
import { NormalizedCacheObject } from 'apollo-cache-inmemory';
import withApollo, { InitApolloOptions } from 'next-with-apollo';
import { getStandaloneApolloClient } from '../utils/graphql';

/**
 * Export a HOC from next-with-apollo
 *
 * Universal, works both on client and server sides
 * Doesn't fetch any data by itself, but provides a client that allows to do it in children components
 *
 * @see https://www.npmjs.com/package/next-with-apollo
 */
export default withApollo(
  ({ initialState }: InitApolloOptions<NormalizedCacheObject>) =>
    getStandaloneApolloClient(initialState), {
    getDataFromTree,
  },
);
``` 

Could you tell me exactly where I use `getInitialProps`? Because I'm not aware of that. (is it a guess on your side maybe?)

**Comment by ijjk** [comment]
@Vadorequest it's being added by the `withApollo` HOC from the `next-with-apollo` package. As mentioned in [the project's docs](https://github.com/lfades/next-with-apollo#using-getinitialprops) if you pass the `getDataFromTree` config it will add `getInitialProps` and opt the pages out of the automatic static optimization

**Comment by Vadorequest** [comment]
Thank you, I wasn't aware of that. This explains a few changes in behaviour I had noticed but hadn't understood the root cause.

Do you think this shows a need for a more detailed warning? Basically, is it technically possible to automatically tell the developer from where come the `getInitialProps`? Like `withApollo` in this example.

**Comment by timneutkens** [comment]
Solved by #12491

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
